flocker/node/agents/blockdevice.py

```diff
@@ -1395,6 +1418,61 @@ def _manifestation_from_volume(volume):
     return Manifestation(dataset=dataset, primary=True)
 
 
+def _ext4_fs_uuid(device_path):
+    """
+    Get the EXT4 filesystem UUID from the filesystem on the device at the given
+    path.
+
+    :param FilePath device_path: The path to the device on which the filesystem
+        resides.
+
+    :return: A ``UUID`` instance giving the UUID from the filesystem.
+    :rtype: ``UUID``
+    """
+    blkid = _blkid(
+        # Display a slightly more parseable format.
+        b"-o", b"udev",
+        # Display information about this device
+        device_path.path,
+    )
+    # Output is lines like:
+    # ID_FS_UUID=a0d8c288-2dc9-4fa8-9039-f834d69730c6
+    # ID_FS_UUID_ENC=a0d8c288-2dc9-4fa8-9039-f834d69730c6
+    # ID_FS_TYPE=ext4
+    for line in blkid.splitlines():
+        key, value = line.split(b"=")
+        if key == b"ID_FS_UUID":
+            return UUID(value)
+    raise ValueError(
+        "Failed to find UUID on {} in output: {}".format(
+            device_path.path, blkid
+        )
+    )
+
+
+def tune2fs(*argv):
+    """
+    Run the ``tune2fs`` utility with the given arguments and return its stdout
+    and stderr.
+
+    :param argv: ``bytes`` giving the argument vector to pass to the process.
+
+    :return: ``bytes`` giving the stdout and stderr of tune2fs (mixed
+        together).
+    """
+    return check_output([b"tune2fs"] + list(argv), stderr=STDOUT)
```

Comment:
:question: And if the command fails, this'll mean that the STDERR is included in the ``CalledProcessError``, right?....which I think is a good thing for debugging if this fails for some reason.